### ISSUE TITLE
rxvt-unicode-pixbuf image previews go out of borders

### ISSUE THREAD

**Issue by raycewest** [ranger/ranger#815]
#### ISSUE TYPE

- Bug report

#### RUNTIME ENVIRONMENT

- Operating system and version: Arch
- Terminal emulator and version: rxvt-unicode-pixbuf from AUR, as suggested in wiki for previews
- Python version: 3.6.0
- Ranger version/commit: 1.8.1-1

#### EXPECTED BEHAVIOR

 Previews image but bound by borders

#### CURRENT BEHAVIOR

See image below.  All images tested seem to align to the far right of the terminal.

#### STEPS TO REPRODUCE

1. install rxvt-unicode-pixbuf, w3m, ranger
2. rc.conf: draw_borders and preview_images set to true, preview_images_method to urxvt
3. ranger --copy-config=scope, then run

#### IMAGES / VIDEOS

![image](https://cloud.githubusercontent.com/assets/10680546/23176273/fae80e10-f827-11e6-95db-dd3633897565.png)

**Comment by nfnty** [comment]
Have you tested running from `master`?

**Comment by raycewest** [comment]
Yes.

**Comment by raycewest** [comment]
FTR, works perfectly with xterm and "set preview_images_method w3m", save for the image just barely covering the border to its left.

**Comment by Vifon** [comment]
Yes, it happens and sadly it probably will keep happening as I have no good ideas how to properly solve it. The placement of the `urxvt` previews is a pretty tricky subject.

**Comment by crocket** [comment]
I'm affected by this issue, too. Image previews hide file names and the bottom status bar.

**Comment by ArtemBan** [comment]
I created a plugin to make urxvt displayer place image relatively correct. It needs w3mimgdisplay installed because I don't know how to get font size properly.

    import sys
    from ranger.ext.img_display import *

    def draw(self, path, start_x, start_y, width, height):
        # The coordinates in the arguments are ignored as urxvt takes
        # the coordinates in a non-standard way: the position of the
        # image center as a percentage of the terminal size. As a
        # result all values below are in percents.
    
        full_h, full_w = self.fm.ui.win.getmaxyx()
        iw, ih = ITerm2ImageDisplayer()._get_image_dimensions(path)

        w3m = W3MImageDisplayer()
        w3m.initialize()
        fw, fh = w3m._get_font_dimensions()
        w3m.quit()
                                        
        scale = min(float(fw*width)/iw, float(fh*height)/ih)
                                                        
        x = int(round(100*fw*start_x / (fw*full_w - iw*scale)))
        y = int(round(100*fh*start_y / (fh*full_h - ih*scale)))
        w = 100*width/full_w
        h = 100*height/full_h
                         
        sys.stdout.write("\033]20;{path};{w}x{h}+{x}+{y}:op=keep-aspect\a".format(**vars()))
        sys.stdout.flush()

    URXVTImageDisplayer.draw = draw

**Comment by raycewest** [comment]
This centers the image across the entire terminal - not the third pane of ranger.  Was this your intent?

**Comment by by-speece** [comment]
2022 - st - problem exist
